hw/ip/otbn/util/rig/program.py

```diff
@@ -299,3 +299,24 @@ def pick_branch_target(self, min_len: int) -> Optional[int]:
 
         assert len(tgts) == 1
         return tgts[0]
+
+    def get_blank_insns_above(self, addr: int) -> int:
+        '''Return the number of instructions that fit above addr'''
+        bytes_above = self.imem_size - addr
+        if bytes_above <= 0:
+            return 0
+
+        for sec_base, sec_insns in self._sections.items():
+            sec_top = sec_base + 4 * len(sec_insns)
```

Comment:
Hmm, maybe I'll go for "end". The problem is that I want to distinguish between "max" (0xfff or similar) and "top" (0x1000). I reckon you've got your number line upside down :-D, but I agree this might be confusing - will rename.